rfcs/0000-c-unwind-and-project-group-charter.md

```diff
@@ -0,0 +1,496 @@
+unwinding. Therefore, attempts to unwind through a `extern "C"`
+barrier are considered [Undefined Behavior]. However, in practice, a
+number of projects have found that "things work fine" so long as
+certain conditions are met (for example, the frames that are to be
+unwound do not contain destructors).
+
+We have made a number of efforts to "tighten up" the behavior around
+the C ABI and unwinding. For example, when users define a Rust
+function that has the C ABI, we would like to make such functions
+abort if unwinding actually occurs at runtime. Unfortunately, these
+attempts have caused existing, working projects to break.
+
+The problem here is not that the existing projects break *per se*: they
+are relying on [Undefined Behavior] and that is to be expected as a
+possibility. The problem is that there is no alternative available to
+them that could allow them to keep working (even if they are
+continuing to rely on behavior that is not yet fully specified).
+
+## Specific use cases
+
+We have seen a number of unwinding use cases in the wild. This section
+details some of those use cases and their specific requirements.
+
+### Lucet: unwind past empty native frames
+
+The **[Lucet]** compiler generates native frames that invoke Rust
+helpers. These frames are always invoked from some Rust helper.  From
+time to time, they may invoke other Rust helpers, resulting in native
+frames "sandwiched" between two Rust frames:
+
+[Lucet]: https://www.fastly.com/blog/announcing-lucet-fastly-native-webassembly-compiler-runtime
+
+```
+[Rust root frame]
+     |
+     v
+[Native frames]
+     |
+     v
+[Rust helper function]
+```
+
+This Rust helper function may panic. Lucet would like to be able to
+catch this panic from the Rust root frame. They would like to have the
+Native frames "gracefully unwind". However, the native frames
+themselves do not contain any destructors or landing pads -- they
+simply need to be "skipped over", in effect.
+
+Moreover, Lucet executes only on a narrow range of platforms:
+
+* XXX details
+
+### mozjpeg: XXX
+
+XXX somebody fill this in XXX
+
+### lua bindings and longjmp
+
+Although longjmp/setjmp are not directly in scope for this group,
+there is an interesting interaction that was uncovered and is worth
+documenting. On Windows targets specifically, longjmp is implemented
+using the same "Structured Exception Handling" (SEH) mechanism that is
+used for C++ exceptions and other sorts of errors. As a result, in
+some versions of Rust at least, we found that attempts to longjmp over
+Rust frames triggered unwinding mechanisms. Note that in general a
+longjmp over a Rust frame that contains destructors is [Undefined
+Behavior] -- however, we would like to support longjmp for Rust frames
+that do not contain destructors. At minimum, this group should take
+the behavior of longjmp into account; if it is easy to do, we may also
+introduce mechanisms or RFCs that help to specify the use of longjmp
+in particular cases.
```

Comment:
So this isn't really that related to `longjmp`, although `longjmp` exhibits this issue. As the text says, on these platforms, these longjmps are implemented using "native unwinds". The problem is that on these platforms "native unwinds" can behave differently, depending on, e.g., how they are thrown. 

If this WG attacks the problem of how native unwinds behave in Rust, it would need to consider those potential differences in behavior. 

I'm not sure how to word that into this section which appears more focused on the issue discovered while using rlua on windows.